python/Ganga/Core/GangaRepository/SessionLock.py

```diff
@@ -286,6 +268,25 @@ def decorated(self, *args, **kwargs):
             return f(self, *args, **kwargs)
     return decorated
 
+def global_disk_lock(f):
+    @functools.wraps(f)
+    def decorated_global(self, *args, **kwds):
+        with global_disk_lock.global_lock:
+            while True:
+                try:
+                    self.global_lock_acquire()
+                    break
+                except:
+                    time.sleep(1.)
+
+            self.safe_LockCheck()
+            try:
+                return f(self, *args, **kwds)
+            finally:
+                self.global_lock_release()
+    return decorated_global
+
+global_disk_lock.global_lock = threading.Lock()
```

Comment:
I was happy to make this a global variable but several people have expressed opinions against global objects.